### PR TITLE
[KDA-Pilot] Add diffusion causal Conv3D cat-pad CUDA fast path for Cosmos3

### PR DESCRIPTION
**Pull request by BBuf**
Powered by [KDA-Pilot](https://github.com/mit-han-lab/kernel-design-agents), we propose a native CUDA JIT fast path for the Cosmos3 VAE causal Conv3D cat/pad copy pattern.

## Motivation

Add a native CUDA JIT fast path for the Cosmos3 VAE causal Conv3D cat/pad copy pattern from KDA-Pilot task `b200_diffusion_causal_conv3d_cat_pad__multi_shape` / [BBuf/KDA-Pilot#132](https://github.com/BBuf/KDA-Pilot/pull/132).

The hot pattern is equivalent to:

```python
F.pad(torch.cat([cache_x, x], dim=2), (Wl, Wr, Ht, Hb, Dl - cache_t, Dr))
```

for contiguous 5D `NCTHW` tensors. A Cosmos3-Nano T2V trace at `832x480x9f, 4 steps` hits 48 fuseable calls across 8 production shapes in VAE decode. This is not a denoise kernel, so expected model-level impact is modest and should show up around decode/total rather than denoise.

## Modifications

- Add `diffusion_causal_conv3d_cat_pad` lightweight JIT CUDA kernel.
- Mark the CUDA csrc file with the MIT HAN Lab Kernel Design Agents provenance comment.
- Register it as an eager SGLang custom op with a fake impl so direct `torch.compile` tracing sees an opaque op instead of tracing JIT/module loading.
- Wire `parallel_conv.causal_conv3d_cat_pad` to try CUDA by default, then fall back to the existing Triton implementation if the JIT load/launch fails.
- Use static Cosmos3 compile shapes (`dynamic=False`) for the compiled gen-layer path; this keeps the Cosmos3 `--enable-torch-compile` model path runnable with the compile-safe custom op.
- Add a concise correctness test over the Cosmos3 shape set plus a direct `torch.compile(fullgraph=True)` custom-op test.
- Add a standalone benchmark for the 8 traced Cosmos3 shapes.

## Accuracy Tests

```bash
python3 -m py_compile \
  python/sglang/jit_kernel/diffusion/causal_conv3d_cat_pad.py \
  python/sglang/multimodal_gen/runtime/layers/parallel_conv.py \
  python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/cosmos3.py \
  test/registered/jit/diffusion/test_causal_conv3d_cat_pad.py \
  test/registered/jit/benchmark/diffusion/bench_causal_conv3d_cat_pad.py
python3 -m ruff format --check ...
python3 -m ruff check ...
git diff --check
```

B200:

```bash
CUDA_VISIBLE_DEVICES=0 \
PYTHONPATH=/tmp/sglang_catpad_pr/python:/tmp/sglang_catpad_pr \
TVM_FFI_CACHE_DIR=/tmp/tvm-ffi-catpad-b200-final \
python3 -m pytest -q test/registered/jit/diffusion/test_causal_conv3d_cat_pad.py -s
```

Result: `9 passed`.

H100:

```bash
CUDA_VISIBLE_DEVICES=0 \
PYTHONPATH=/tmp/sglang_catpad_pr/python:/tmp/sglang_catpad_pr \
TVM_FFI_CACHE_DIR=/tmp/tvm-ffi-catpad-h100-final \
python3 -m pytest -q test/registered/jit/diffusion/test_causal_conv3d_cat_pad.py -s
```

Result: `9 passed`.

## Speed Tests and Profiling

Kernel benchmark command:

```bash
CUDA_VISIBLE_DEVICES=0 \
PYTHONPATH=/tmp/sglang_catpad_pr/python:/tmp/sglang_catpad_pr \
TVM_FFI_CACHE_DIR=/tmp/tvm-ffi-catpad-<gpu> \
python3 test/registered/jit/benchmark/diffusion/bench_causal_conv3d_cat_pad.py
```

B200 kernel benchmark:

| Case | Trace calls | Triton us | CUDA us | Speedup |
| --- | ---: | ---: | ---: | ---: |
| c1024_t1_h30_w52_cache1 | 8 | 19.072 | 14.336 | 1.33x |
| c1024_t1_h30_w52_cache2 | 8 | 26.624 | 19.456 | 1.37x |
| c1024_t2_h60_w104_cache1 | 5 | 81.920 | 48.032 | 1.71x |
| c1024_t2_h60_w104_cache2 | 5 | 109.568 | 61.440 | 1.78x |
| c512_t4_h120_w208_cache1 | 5 | 250.688 | 126.976 | 1.97x |
| c512_t4_h120_w208_cache2 | 5 | 306.176 | 150.528 | 2.03x |
| c256_t4_h240_w416_cache1 | 6 | 487.392 | 231.392 | 2.11x |
| c256_t4_h240_w416_cache2 | 6 | 598.272 | 274.400 | 2.18x |

Weighted by traced call count: **10.621 ms -> 5.240 ms**, **2.03x**, saves **5.381 ms** in the kernel group.

H100 kernel benchmark:

| Case | Trace calls | Triton us | CUDA us | Speedup |
| --- | ---: | ---: | ---: | ---: |
| c1024_t1_h30_w52_cache1 | 8 | 18.752 | 15.296 | 1.23x |
| c1024_t1_h30_w52_cache2 | 8 | 28.192 | 19.456 | 1.45x |
| c1024_t2_h60_w104_cache1 | 5 | 90.336 | 50.976 | 1.77x |
| c1024_t2_h60_w104_cache2 | 5 | 118.560 | 65.888 | 1.80x |
| c512_t4_h120_w208_cache1 | 5 | 279.744 | 137.856 | 2.03x |
| c512_t4_h120_w208_cache2 | 5 | 334.816 | 164.160 | 2.04x |
| c256_t4_h240_w416_cache1 | 6 | 546.336 | 251.520 | 2.17x |
| c256_t4_h240_w416_cache2 | 6 | 659.616 | 300.352 | 2.20x |

Weighted by traced call count: **11.729 ms -> 5.684 ms**, **2.06x**, saves **6.045 ms** in the kernel group.

B200 model E2E benchmark with `torch.compile` enabled:

- Host: `ion-b200`, NVIDIA B200, `CUDA_VISIBLE_DEVICES=7`.
- Model path: `nvidia/Cosmos3-Nano`, native `Cosmos3Pipeline`; no diffusers fallback observed.
- Shape: T2V `832x480x9f`, `num_inference_steps=4`, `seed=42`, `save_output=False`.
- Compile path: `enable_torch_compile=True`, Cosmos3 gen layers compiled with `{'mode': 'default', 'fullgraph': False, 'dynamic': False}`.
- Measurement: one loaded local `DiffGenerator` process per implementation, 4 warmup requests, median of 15 measured requests. The baseline forces the existing Triton fallback; the PR row uses the default CUDA fast path. Metric is `GenerationResult.metrics.total_duration_ms`.

| Implementation | Median E2E ms | Mean E2E ms | Min / Max E2E ms | Speedup |
| --- | ---: | ---: | ---: | ---: |
| Triton fallback | 181.521 | 181.388 | 176.617 / 187.432 | 1.00x |
| CUDA JIT fast path | 177.687 | 177.554 | 173.898 / 180.383 | **1.021x (+2.11%)** |

## B200 Visual Output Check

Visual equivalence check uses eager generation (`--enable-torch-compile false`) to compare the current `main` Triton path with this PR's default CUDA fast path. Performance evidence above uses the `torch.compile` path.

- Host/GPU: `ion-b200`, NVIDIA B200, `CUDA_VISIBLE_DEVICES=7`.
- Main commit: `e4976683f4`; PR commit: `869bad2b1053`.
- Prompt/shape: `A blue box slides across a clean warehouse floor.`, `832x480x9f`, `num_inference_steps=4`, `seed=42`.
- Native `Cosmos3Pipeline` was used for both runs; no diffusers fallback was observed.
- Decoded-video comparison: SSIM All `1.000000`, PSNR `inf`.

Left is `main`; right is this PR:

Cosmos3 main vs PR visual comparison

MP4 artifacts: [main.mp4](https://raw.githubusercontent.com/BBuf/sglang/pr29281-assets/cosmos3/main.mp4), [pr.mp4](https://raw.githubusercontent.com/BBuf/sglang/pr29281-assets/cosmos3/pr.mp4).

## Checklist

- [x] Default path tries CUDA first and falls back to Triton on failure.
- [x] No environment variable is required to enable the CUDA kernel.
- [x] Direct custom op registration is eager and has a fake impl for `torch.compile` compatibility.
- [x] Cosmos3 `--enable-torch-compile` model path tested on B200 with `dynamic=False` compile shapes.
- [x] B200 kernel benchmark and unit test included.
- [x] H100 kernel benchmark and unit test included.
- [x] Cosmos3 native model smoke tested on B200 and H100.

## Review and Merge Process

This is a focused KDA-Pilot kernel integration PR. The expected useful effect is the 2x faster VAE cat/pad kernel group; whole-model speedup is intentionally modest because the group is only a small slice of Cosmos3 T2V runtime.

---
### CI States

Latest PR Test (Base): :hourglass_flowing_sand: [Run #28213972400](https://github.com/sgl-project/sglang/actions/runs/28213972400)
Latest PR Test (Extra): :x: [Run #28213972315](https://github.com/sgl-project/sglang/actions/runs/28213972315)

### CODE REVIEW

**Comment by BBuf** [comment]
Merged with all ci green. https://github.com/sgl-project/sglang/actions/runs/28213972400/job/83581027046?pr=29281
